src/traffic-collector/azext_traffic_collector/tests/latest/test_traffic_collector.py

```diff
@@ -0,0 +1,93 @@
+                     self.check('@[0].resourceGroup', self.kwargs['rg']),
+                     self.check('@[0].name', self.kwargs['name'])]
+                 )
+
+        self.cmd('az network-function traffic-collector show --resource-group {rg} --name {name}',
+                 checks=[
+                     self.check('name', self.kwargs['name']),
+                     self.check('resourceGroup', self.kwargs['rg']),
+                     self.check('type', 'Microsoft.NetworkFunction/AzureTrafficCollectors'),
+                 ])
+
+        self.cmd(
+            'az network-function traffic-collector update --resource-group {rg} --name {name} --tags foo=doo1',
+            checks=[
+                self.check('name', self.kwargs['name']),
+                self.check('resourceGroup', self.kwargs['rg']),
+                self.check('tags.foo', 'doo1')
+            ])
+
+        # Collector Policy Tests
+        self.cmd('az network-function traffic-collector collector-policy create --resource-group {rg} '
+                 '--azure-traffic-collector-name {name} -n {cp_name} -l {location} --ingestion-policy {ip}',
+                 checks=[
+                     self.check('name', self.kwargs['cp_name'])
+                 ])
+
+        self.cmd('az network-function traffic-collector collector-policy show --resource-group {rg} '
+                 '--azure-traffic-collector-name {name} -n {cp_name}',
+                 checks=[
+                     self.check('name', self.kwargs['cp_name']),
+                     self.check('emissionPolicies | length(@)', 0),
+                     self.check('ingestionPolicy.ingestionSources | length(@)', 1),
+                     self.check('type', 'Microsoft.NetworkFunction/AzureTrafficCollectors/CollectorPolicies')
+                 ])
+
+        self.cmd('az network-function traffic-collector collector-policy update --resource-group {rg} '
+                 '--azure-traffic-collector-name {name} -n {cp_name} -l {location} --emission-policies [0]={ep}',
+                 checks=[
+                     self.check('name', self.kwargs['cp_name'])
+                 ])
+
+        self.cmd('az network-function traffic-collector collector-policy show --resource-group {rg} '
+                 '--azure-traffic-collector-name {name} -n {cp_name}',
+                 checks=[
+                     self.check('name', self.kwargs['cp_name']),
+                     self.check('emissionPolicies | length(@)', 1),
+                     self.check('ingestionPolicy.ingestionSources | length(@)', 1),
+                     self.check('type', 'Microsoft.NetworkFunction/AzureTrafficCollectors/CollectorPolicies')
+                 ])
+
+        self.cmd(
+            'az network-function traffic-collector collector-policy delete --resource-group {rg} '
+            '--azure-traffic-collector-name {name} --name {cp_name} --yes')
+
+        time.sleep(120)
```

Comment:
Could you talk more about the 2 minutes waiting here?

Reply:
_Background:_ Traffic collector policy needs to be deleted first. Once that operation is finished, then only we can delete the traffic collector. 

I added timer because traffic-collector delete was getting executed right after the collector-policy delete command when collector-policy delete command was still being processed. This lead to test failure giving error similar to "can't delete traffic collector with an active collector policy". That's why I added timer.